build/pause/Makefile

```diff
@@ -85,8 +85,7 @@ push-manifest:
 	set -x; \
 	for arch in $(ALL_ARCH.windows);  do \
 		for osversion in ${ALL_OSVERSIONS.windows}; do \
-			BASEIMAGE=${BASE.windows}:$${osversion}; \
-			full_version=`docker manifest inspect ${BASE.windows}:$${osversion} | grep "os.version" | head -n 1 | awk -F\" '{print $$4}'` || true; \
+			full_version=`docker manifest inspect mcr.microsoft.com/windows/nanoserver:$${osversion} | grep "os.version" | head -n 1 | awk -F\" '{print $$4}'` || true; \
```

Comment:
Is it expected that this is hardcoded to `mcr.microsoft.com/windows/nanoserver`? Why don't we use `${BASE}` here?

Reply:
One possibility might be that the `${BASE}` image might not have the `os.version` data set (or set correctly).
Some build processes may not set it like this code is doing but the nanoserver image should always have it set.
(Just a guess...)

Reply:
It's because we can't assume that the windows base image has the same image tag as the custom base image we want to inject. That's why I use nanoserver as the source of truth for inspecting the windows os.version. Let me see if I can override this as well.

Reply:
I can't think of a generic way to grab the `os.version` field because the current Makefile assumes that the base image manifest list only contains one `os.version` entry. @marosset is it safe to use the latest `os.version` even if the custom windows base image is out of date (e.g. `mcr.microsoft.com/windows/nanoserver:1809` returns os.version `10.0.17763.2458` but the custom windows base image returns `10.0.17763.2452`)?

Reply:
> I can't think of a generic way to grab the `os.version` field because the current Makefile assumes that the base image manifest list only contains one `os.version` entry. @marosset is it safe to use the latest `os.version` from nanoserver even if the custom windows base image is out of date (e.g. `mcr.microsoft.com/windows/nanoserver:1809` returns os.version `10.0.17763.2458` but the custom windows base image returns `10.0.17763.2452`)?

No, we shouldn't use the latest os.version if the base image isn't also using the same version.

Reply:
The patch level is used to identify which cumulative security updates have been applied to the container images.
We should not misrepreent this information.